pkg/fetch/fetch.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fetch
```

Comment:
Unless you expect to use this outside of apko, it might worth putting this in an `internal` package, so it doesn't grow external dependencies. (That would also obviate all my unexporting comments)

Reply:
I plan to use it in Melange too for pipelines! :)